codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Decoder.java

```diff
@@ -526,32 +529,29 @@ private boolean exceedsMaxHeaderSize(long size) {
         return true;
     }
 
-    private byte[] readStringLiteral(InputStream in, int length) throws IOException {
-        byte[] buf = new byte[length];
-        if (in.read(buf) != length) {
-            throw DECOMPRESSION_EXCEPTION;
-        }
-
+    private CharSequence readStringLiteral(ByteBuf in, int length) throws IOException {
         if (huffmanEncoded) {
-            return Huffman.DECODER.decode(buf);
+            return huffmanDecoder.decode(in, length);
         } else {
-            return buf;
+            byte[] buf = new byte[length];
+            in.readBytes(buf);
+            return new AsciiString(buf, false);
```

Comment:
Question: does AsciiString imply the contents of the string are Ascii characters (0-127), or that each byte is representative of a single character?  I ask because I had been under the assumption that AsciiString was used only when we know the string encoding was asctually Ascii, and does not include the later characters.  

The hpack spec says "string" but they can really be arbitrary byte sequences.  Only in the context of http are they constrained to Ascii values.